Docker-dev-build

```diff
@@ -0,0 +1,135 @@
+
+# Any extra packages to be installed in the host
+ARG EXTRA_PACKAGES
+RUN echo "Installing EXTRA_PACKAGES ${EXTRA_PACKAGES} on container host"
+
+# general environment for docker
+ENV SPACK_ROOT=/home/user/spack \
+	  SPACK=/home/user/spack/bin/spack \
+	  FORCE_UNSAFE_CONFIGURE=1
+
+RUN apt-get -y update
+# Convenience tools, if desired for debugging etc
+# RUN apt-get -y install wget time nano vim emacs git
+
+# From https://github.com/ax3l/dockerfiles/blob/master/spack/base/Dockerfile:
+# install minimal spack dependencies
+RUN apt-get install -y --no-install-recommends \
+              autoconf \
+              build-essential \
+              ca-certificates \
+              coreutils \
+              curl \
+              environment-modules \
+	            file \
+              gfortran \
+              git \
+              openssh-server \
+              python \
+              unzip \
+              vim \
+              wget \
+           && rm -rf /var/lib/apt/lists/*
+
+# load spack environment on login
+RUN echo "source $SPACK_ROOT/share/spack/setup-env.sh" \
+           > /etc/profile.d/spack.sh
+
+RUN adduser user
+USER user
+WORKDIR /home/user
+
+# install spack
+RUN git clone https://github.com/spack/spack.git
+# default branch is develop
+RUN cd spack && git checkout $SPACK_VERSION
+
+# # show which version we use
+RUN $SPACK --version
+
+# copy our package.py into the spack tree (and also example files)
+COPY spack/package.py $SPACK_ROOT/var/spack/repos/builtin/packages/octopus/package.py
+RUN ls -l $SPACK_ROOT/var/spack/repos/builtin/packages/octopus
+COPY spack/test/ $SPACK_ROOT/var/spack/repos/builtin/packages/octopus/test
+RUN ls -l $SPACK_ROOT/var/spack/repos/builtin/packages/octopus/test
+COPY check_buildlog.py /home/user 
+
+# Install and test serial and MPI versions of ocoptus via spack
+# # serial version
+
+RUN . $SPACK_ROOT/share/spack/setup-env.sh && \
+      # create a new environment for the serial version and activate it:
+      spack env create octopus-serial && \
+      spack env activate octopus-serial && \
+      # make a dev-build direcotry for the serial version:
+      mkdir ~/dev-build-serial && cd ~/dev-build-serial && \
+
+      wget https://octopus-code.org/down.php?file=12.1/octopus-12.1.tar.gz -O octopus-12.1.tar.gz &&\
+      tar --strip-components=1 -xf octopus-12.1.tar.gz -C ./&& \
+      # install the serial version of octopus:
+      # display specs of upcoming spack installation:
+      spack spec octopus@${OCT_VERSION} ~mpi+netcdf+parmetis+arpack+cgal+pfft+python+likwid+libyaml+elpa+nlopt~debug~cuda~metis~scalapack && \
+      # run the spack installation (adding it to the environment):
+      spack dev-build octopus@${OCT_VERSION} ~mpi+netcdf+parmetis+arpack+cgal+pfft+python+likwid+libyaml+elpa+nlopt~debug~cuda~metis~scalapack && \
+      # run spack smoke tests for octopus. We get an error if any of the fails:
+      spack test run --alias test_serial octopus && \
```

Comment:
Comment B: What happens here? If octopus is not installed yet, do we install it here? Or is this line exceuted without effect because the package is not installed yet?

I would like to avoid installing Octopus via spack here (and would rather skip the smoke tests if that is the choice).

Reply:
As in the reply to the previous comment, octopus is actually installed and the spacks view is updated with it. These likes actually use the spack installed octopus to run smoke tests